libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/camera/CameraComponent.kt

```diff
@@ -126,8 +128,10 @@ internal class CameraComponent constructor(
         }
     }
 
+    @OptIn(FlowPreview::class)
     private fun syncNavigationCameraState() {
-        navigationCamera.flowNavigationCameraState().observe {
+        // debounce is necessary to skip intermediate IDLE state when transitioning to FOLLOWING
+        navigationCamera.flowNavigationCameraState().debounce(10).observe {
             store.dispatch(SetCameraMode(it.toTargetCameraMode()))
```

Comment:
Commenting for another suggestion here. What if the camera state was removed from the store `State`

That way the camera mode does not trigger a potential infinite loop